shared/libraries/config_protocol/include/config_protocol/config_client_device_impl.h

```diff
@@ -123,6 +129,35 @@ uint64_t GenericConfigClientDeviceImpl<TDeviceBase>::onGetTicksSinceOrigin()
     return ticks;
 }
 
+template <class TDeviceBase>
+ListPtr<IDeviceInfo> GenericConfigClientDeviceImpl<TDeviceBase>::onGetAvailableDevices()
+{
+    return List<IDeviceInfo>();
+}
+
+template <class TDeviceBase>
+DictPtr<IString, IDeviceType> GenericConfigClientDeviceImpl<TDeviceBase>::onGetAvailableDeviceTypes()
+{
+    return Dict<IString, IDeviceType>();
+}
+
+template <class TDeviceBase>
+DevicePtr GenericConfigClientDeviceImpl<TDeviceBase>::onAddDevice(const StringPtr& /*connectionString*/, const PropertyObjectPtr& /*config*/)
+{
+    throw NotImplementedException{};
```

Comment:
Just to clarify: does this mean that adding a remote device in the same way as adding a function block via the native config protocol is completely disallowed, or is this just a temporary limitation due to the current lack of implementation?

Reply:
This is a temporary limitation. It should work and invoke an RPC, but NotImplemented better represents the current state than some arbitrary default behaviour.